### PR TITLE
enum-flags: add 0.1a

### PR DESCRIPTION
**Pull request by theodelrieu**
Specify library name and version:  **enum-flags/0.1a**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/enum-flags/0.1a/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    default_options = "disable_implicit_conversion=True"
+    generators = "cmake"
+    _source_subfolder = "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def build(self):
```
Comment (danimtb) [inline recipes/enum-flags/0.1a/conanfile.py]: Remove this method

File: recipes/enum-flags/0.1a/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        pass
+
+    def package(self):
+        self.copy("include*", src=self._source_subfolder)
+        self.copy("*LICENSE", dst="licenses", keep_path=False)
+
+    def package_info(self):
+        # Yes, there is a typo in the macro name.
```
Comment (danimtb) [inline recipes/enum-flags/0.1a/conanfile.py]: Thanks for the clarification!

File: recipes/enum-flags/0.1a/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+class EnumFlagsConan(ConanFile):
+    name = "enum-flags"
+    description = "Bit flags for C++11 scoped enums"
+    homepage = "https://github.com/grisumbras/enum-flags"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("bitmask", "enum")
+    license = "MIT"
+    options = {"disable_implicit_conversion": [True, False]}
+    default_options = "disable_implicit_conversion=True"
```
Comment (danimtb) [inline recipes/enum-flags/0.1a/conanfile.py]: I always have trouble with the double negations. Can we change it to `enable_*` or is it a documented name for this option?

File: recipes/enum-flags/0.1a/test_package/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if self.settings.os == "Emscripten":
+            ext = ".js"
+        elif tools.os_info.is_windows:
+            ext = ".exe"
+        else:
+            ext = ""
```
Comment (danimtb) [inline recipes/enum-flags/0.1a/test_package/conanfile.py]: Can we simplify this logic a little bit?

File: recipes/enum-flags/0.1a/conanfile.py (hunk 3 above)
Comment (theodelrieu) [reply recipes/enum-flags/0.1a/conanfile.py]: Well, the macro name contains `FORBID`, so I figured it would be consistent with the library's choice.

Comment (madebr) [reply recipes/enum-flags/0.1a/conanfile.py]: Also, `default_options` should be a dict.

Comment (danimtb) [reply recipes/enum-flags/0.1a/conanfile.py]: I would use the same then `forbid_implicit_conversion`

File: recipes/enum-flags/0.1a/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+class EnumFlagsConan(ConanFile):
+    name = "enum-flags"
+    description = "Bit flags for C++11 scoped enums"
+    homepage = "https://github.com/grisumbras/enum-flags"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("bitmask", "enum")
+    license = "MIT"
+    options = {"forbid_implicit_conversions": [True, False]}
+    default_options = {"forbid_implicit_conversions": True}
+    generators = "cmake"
```
Comment (uilianries) [inline recipes/enum-flags/0.1a/conanfile.py]:
```suggestion
    generators = "cmake"
    no_copy_source = True
```
https://docs.conan.io/en/latest/reference/conanfile/attributes.html#no-copy-source

File: recipes/enum-flags/0.1a/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
+from conans import ConanFile, tools
+import os
+
+
+class EnumFlagsConan(ConanFile):
+    name = "enum-flags"
```
Comment (uilianries) [inline recipes/enum-flags/0.1a/conanfile.py]:
```suggestion
    name = "enum-flags"
    version = "0.1a"
```
As you are using a specific folder, you can set a specific version here.

Comment (theodelrieu) [reply recipes/enum-flags/0.1a/conanfile.py]: Is it really required? I will have to remove it if a second version gets released (I did the same in all the PRs I've opened)

Comment (SSE4) [reply recipes/enum-flags/0.1a/conanfile.py]: I don't think it's required - we may add more versions, so let's keep as is

**Pull request merged**